### ISSUE TITLE
ANR Triggered by InstabugAnnouncementSubmitterService

### ISSUE THREAD

**Issue by plowman** [Instabug/Instabug-Android#226]
### What I'm seeing
<img width="620" alt="instabug_anr" src="https://user-images.githubusercontent.com/148066/64901428-f43d6180-d64d-11e9-88d1-829f9791c41a.png">

Right now in the "ANR" view of the Google Play Console, it seems like basically all of our users are seeing this problem where the app freezes when it starts and an Instabug service is to blame. **Even stranger**: I only see this issue when releasing with an Android bundle rather than an APK.

### Steps to Reproduce the Problem
1. Install the Android bundletool
```
wget https://github.com/google/bundletool/releases/download/0.10.2/bundletool-all-0.10.2.jar
alias bundletool="java -jar bundletool-all-0.10.2.jar"
```
2. Open an emulator or plug in your Android device.
3. Bundle your app for release on Google Play using an Android bundle (.aar) instead of an APK (.apk). Via the command line it's something like this:
```
# From RN root directory
cd android
./gradlew bundleProductionRelease
bundletool build-apks --connected-device --bundle=./app/build/outputs/bundle/productionRelease/app-production-release.aab --output="./production-release.apks"
```
4. Install the app
```
bundletool install-apks --apks="./production-release.apks"
```
5. Open the app.

### Expected Behavior
* It should open

### Actual Behavior
* It hangs forever, until an "Application Not Responding" is shown
* In the logs I see this: `09-13 17:33:03.097   899   899 W JobServiceContext: Time-out while trying to bind 15ca94f #u0a242/2583 com.myapp.android/com.instabug.survey.announcements.network.InstabugAnnouncementSubmitterService, dropping.`

### Instabug integration code
```
// MainApplication.java
    protected List<ReactPackage> getPackages() {
        List<ReactPackage> packages = new PackageList(this).getPackages();

        packages.add(new RNInstabugReactnativePackage.Builder(BuildConfig.INSTABUG_API_KEY, MainApplication.this)
            .setInvocationEvent("none")
            .build());

        return packages;
    }
```

```
// App.js
  async componentDidMount() {
    Instabug.start(Constants.instabugToken(), [Instabug.invocationEvent.none]);
    BugReporting.setAutoScreenRecordingEnabled(false);
    Instabug.setTrackUserSteps(true);
  }
```

### SDK Version
`"instabug-reactnative": "8.6.3",`

### React Native, iOS and Android Versions
```
  Binaries:
    Node: 12.9.1 - /usr/local/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.11.3 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 12.4, macOS 10.14, tvOS 12.4, watchOS 5.3
    Android SDK:
      API Levels: 23, 26, 27, 28
      Build Tools: 25.0.3, 27.0.3, 28.0.2, 28.0.3
      System Images: android-26 | Google APIs Intel x86 Atom, android-28 | Google Play Intel x86 Atom
  IDEs:
    Android Studio: 3.5 AI-191.8026.42.35.5791312
    Xcode: 10.3/10G8 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.8.6 => 16.8.6 
    react-native: 0.60.5 => 0.60.5 
  npmGlobalPackages:
    react-native-cli: 2.0.1
    react-native-create-library: 3.1.2
```

### In conclusion
Have you seen anything like this before? I wasted basically all day trying to figure out why creating a bundle makes the app freeze while installing via an APK does not have the issue. Happy to help you debug further. Thanks!

**Comment by amrbrkt** [comment]
@plowman would you please share with us the full stack trace of the crash?

**Comment by plowman** [comment]
@amrbrkt since it's an "Application Not Responding" error and not a crash, I don't get a good stack trace, just a stack trace of the main thread when the "App isn't responding [close app] [wait]" dialog shows up.

Here's the stack trace of the main thread at that time, though it's doing React Native stuff so I'm not sure if this will be helpful:
```
  at java.lang.Object.wait (Native method)
- waiting on <0x02d54491> (a java.lang.Object)
  at java.lang.Thread.parkFor$ (Thread.java:2137)
- locked <0x02d54491> (a java.lang.Object)
  at sun.misc.Unsafe.park (Unsafe.java:358)
  at java.util.concurrent.locks.LockSupport.park (LockSupport.java:190)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt (AbstractQueuedSynchronizer.java:868)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireQueued (AbstractQueuedSynchronizer.java:900)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire (AbstractQueuedSynchronizer.java:1223)
  at java.util.concurrent.locks.ReentrantReadWriteLock$WriteLock.lock (ReentrantReadWriteLock.java:950)
  at com.facebook.soloader.SoLoader.doLoadLibraryBySoName (SoLoader.java:702)
  at com.facebook.soloader.SoLoader.loadLibraryBySoName (SoLoader.java:591)
- locked <0x08d88ff6> (a java.lang.Object)
  at com.facebook.soloader.SoLoader.loadLibraryBySoName (SoLoader.java:539)
  at com.facebook.soloader.DirectorySoSource.loadDependencies (DirectorySoSource.java:103)
  at com.facebook.soloader.DirectorySoSource.loadLibraryFrom (DirectorySoSource.java:73)
  at com.facebook.soloader.UnpackingSoSource.loadLibrary (UnpackingSoSource.java:465)
- locked <0x0e566ff7> (a java.lang.Object)
  at com.facebook.soloader.SoLoader.doLoadLibraryBySoName (SoLoader.java:681)
  at com.facebook.soloader.SoLoader.loadLibraryBySoName (SoLoader.java:591)
- locked <0x00f51e64> (a java.lang.Object)
  at com.facebook.soloader.SoLoader.loadLibrary (SoLoader.java:529)
  at com.facebook.soloader.SoLoader.loadLibrary (SoLoader.java:484)
  at com.facebook.react.ReactInstanceManagerBuilder.getDefaultJSExecutorFactory (ReactInstanceManagerBuilder.java:304)
  at com.facebook.react.ReactInstanceManagerBuilder.build (ReactInstanceManagerBuilder.java:279)
  at com.reactnativenavigation.react.NavigationReactNativeHost.createReactInstanceManager (NavigationReactNativeHost.java:99)
  at com.facebook.react.ReactNativeHost.getReactInstanceManager (ReactNativeHost.java:41)
  at com.reactnativenavigation.react.ReactGateway.<init> (ReactGateway.java:28)
  at com.myapp.MainApplication.createReactGateway (MainApplication.java:40)
  at com.reactnativenavigation.NavigationApplication.onCreate (NavigationApplication.java:28)
  at com.myapp.MainApplication.onCreate (MainApplication.java:28)
  at android.app.Instrumentation.callApplicationOnCreate (Instrumentation.java:1154)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:5882)
  at android.app.ActivityThread.access$1100 (ActivityThread.java:200)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1651)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:193)
  at android.app.ActivityThread.main (ActivityThread.java:6680)
  at java.lang.reflect.Method.invoke (Native method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:858)
```

**Comment by plowman** [comment]
Good news: I thought the fact that the Google Play Console was blaming Instabug for this was solid enough proof that it was at least partially related to Instabug. However, I just fully removed Instabug from my app and it's still happening. There's a **_small_** chance Instabug and another package are both making the same mistake, but it's more likely this is React Native bug or a bug in another library. Closing this.

**Comment by bwoodlt** [comment]
@plowman I'm having the same issue... how did you fix this?
